### PR TITLE
Support Column.index

### PR DESCRIPTION
**Pull request by yongjhih**
#110

### CODE REVIEW

**Comment by joshuapinter** [comment]
Can you give an example of how you would set the index?

**Comment by yongjhih** [comment]
It's implemented for the #110.

For example:

``` java
@Column(name = "timestamp", index = true)
private Date timestamp; 
```

**Comment by joshuapinter** [comment]
Awesome, thanks. Nice work with the pull request too. 

Would love to have a maintainer step up and help @pardom out with the fixing issues and merging pull requests.

**Comment by yongjhih** [comment]
I hope too, and you should be the guy, thanks.

File: src/com/activeandroid/util/SQLiteUtils.java [hunk 1]
```diff
@@ -108,6 +108,38 @@ public static void execSql(String sql, Object[] bindArgs) {
...
+		for (Field field : tableInfo.getFields()) {
+			String definition = createIndexColumnDefinition(tableInfo, field);
+			if (!TextUtils.isEmpty(definition)) {
+				definitions.add(definition);
+			}
+		}
+		if (definitions.isEmpty()) return null;
+
+		return String.format("CREATE INDEX IF NOT EXISTS %s on %s(%s);",
+				"index_" + tableInfo.getTableName(),
```
Comment (vonloxley) [inline src/com/activeandroid/util/SQLiteUtils.java:123]: Perhaps the index-name shouldn’t be that static. I didn’t have time to test this, but does this work with more than one index definition per table?

**Comment by yongjhih** [comment]
1. The new methods of
   `public static String createIndexDefinition(TableInfo tableInfo)`
   and
   `public static String createIndexColumnDefinition(TableInfo tableInfo, Field field)`
   imitated
   `public static String createTableDefinition(TableInfo tableInfo)`
   and
   `public static String createColumnDefinition(TableInfo tableInfo, Field field)`, I think the static index-name that's ok.
2. It's work with more than one index definition per table. For exmaple:

Item.java:

``` java
    @Column(name = NAME, index = true)
    public String name;

    @Column(name = CATEGORY, index = true)
    public String category;
```

Here is the database dumping of index_items:

```
CREATE INDEX index_items on items(name,category)
```

**Comment by vonloxley** [comment]
Ah, sorry. I somehow missed for-loop above. No hard feelings. Thanks a lot for the patch!

**Comment by shreyanschowdhary** [comment]
Hey ! So I have something like this:
@Column(name = "Survey_Id", index = true)
  @SerializedName("id")
    public int mId;
How do I make a query based on the value of the id. 
I tried: 
   return  new Select()
                .from(SurveySpec.class)
                .where("Survey_Id = ?" , 163)
                .executeSingle();

But it returns a null object despite an object with mId = 163 existing in the database. Any help would be appreciated. Thanks !

@yongjhih

**Pull request merged**